app/javascript/pages/Affiliates/Index.tsx

```diff
@@ -334,60 +260,56 @@ const AffiliateRequestsTable = ({
 const formattedSalesVolumeAmount = (amountCents: number) =>
   formatPriceCentsWithCurrencySymbol("usd", amountCents, { symbolFormat: "short" });
 
-export type SortKey = "affiliate_user_name" | "products" | "fee_percent" | "volume_cents";
-export type Params = {
-  page: number | null;
-  query: string | null;
-  sort: Sort<SortKey> | null;
-};
-
-const AffiliatesTab = () => {
-  const navigation = useNavigation();
-  const revalidator = useRevalidator();
+export default function AffiliatesIndex() {
+  const props = usePage<Props>().props;
   const loggedInUser = useLoggedInUser();
-  const [searchParams, setSearchParams] = useSearchParams();
+  const [isNavigating, setIsNavigating] = React.useState(false);
```

Comment:
We already have a `useRouteLoading` hook defined in our app for this. Can you use it and see if it works?
```jsx
import useRouteLoading from "$app/components/useRouteLoading";

// [...]
const isNavigating = useRouteLoading();
```

Then you shouldn't need `setIsNavigating` anymore